editoast/src/modelsv2/infra.rs

```diff
@@ -257,6 +266,58 @@ impl Infra {
             .await?;
         Ok(speed_limits_tags)
     }
+
+    pub async fn get_splited_track_section_with_data(
+        &self,
+        conn: &mut DbConnection,
+        track: Identifier,
+        distance_fraction: f64,
+    ) -> Result<Vec<SplitedTrackSectionWithData>> {
```

Comment:
```suggestion
    ) -> Result<Option<SplitedTrackSectionWithData>> {
```

The `Vec` is not required here since we'll have at most one object (if we have zero it means that the cache is outdated, but it's up to the parent function to panic or handle the error).

Reply:
Done.